### PR TITLE
[NEW] New Message Parser

### PR DESCRIPTION
**Pull request by ggazzo**
## Proposed changes (including videos or screenshots)

The objective is to put an end to the confusion that we face having multiple parsers, and the problems that this brings, it is still experimental then users need to choose to use it.

The benefits are multiple. no more unexpected cases or grammatical collisions (in addition to more flexible nested cases like bold within link labels).
Besides, we no longer render raw html, instead we use components, so the xss attacks are over (the easy ones at least). Without further discoveries and at the fronted, we only reder what is delivered thus improving our performance.
  This can be used in multiple places, (message, alert, sidenav and in the entire mobile application.)

### CODE REVIEW

File: app/lib/server/functions/sendMessage.js [hunk 1]
```diff
@@ -215,6 +216,11 @@ export const sendMessage = function(user, message, room, upsert = false) {
 	parseUrlsInMessage(message);
 
 	message = callbacks.run('beforeSaveMessage', message, room);
+	try {
+		message.md = parser(message.msg);
+	} catch (e) {
+		console.log(e); // temporary
```
Comment (tassoevan) [inline app/lib/server/functions/sendMessage.js:222]: What would be better here?

File: app/lib/server/functions/updateMessage.js [hunk 2]
```diff
@@ -44,6 +45,12 @@ export const updateMessage = function(message, user, originalMessage) {
 
 	message = callbacks.run('beforeSaveMessage', message);
 
+	try {
+		message.md = parser(message.msg);
+	} catch (e) {
+		console.log(e); // temporary
```
Comment (tassoevan) [inline app/lib/server/functions/updateMessage.js:51]: What would be better here?

File: app/lib/server/startup/settings.js [hunk 3]
```diff
@@ -517,6 +517,18 @@ settings.addGroup('Accounts', function() {
 			public: true,
 			i18nLabel: 'Notifications_Sound_Volume',
 		});
+
+		this.add('Accounts_Default_User_Preferences_muteFocusedConversations', true, {
+			type: 'boolean',
+			public: true,
+			i18nLabel: 'Mute_Focused_Conversations',
+		});
```
Comment (tassoevan) [inline app/lib/server/startup/settings.js:525]: ?

File: client/components/Emoji.js [hunk 4]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 
 import { renderEmoji } from '../../app/emoji/client/index';
 
-function Emoji({ emojiHandle }) {
+function Emoji({ emojiHandle, className = undefined }) {
 	const markup = { __html: `${renderEmoji(emojiHandle)}` };
-	return <div dangerouslySetInnerHTML={markup}></div>;
+	return <span className={className} dangerouslySetInnerHTML={markup} />;
```
[the remark this answers is not in the document]
Comment (ggazzo) [reply client/components/Emoji.js:7]: image

**Review by tassoevan — DISMISSED** [review]

**Comment by cb3inco** [comment]
@sampaiodiego and @ggazzo would this have broken webhook POST messages that didn't have content-type defined in the message? One tool we have that uses Slack Notification stopped working in 3.15 and it appears that the content-type was not defined, thus breaking the webhook. 

#23208 - which is what I reference here.

**Pull request merged**